### PR TITLE
Opus encoder outputting non-monotonic PTS

### PR DESCRIPTION
**Pull request by bartkrak**
related to: https://github.com/membraneframework/membrane_core/issues/834

### CODE REVIEW

**Review by FelonEkonom — CHANGES_REQUESTED** [review]
Refresh me when you implement the changes that we spoke about in the office

**Comment by FelonEkonom** [comment]
> @FelonEkonom Dialyzer fails on the part of code that didn't change in this pr, Do you know how to fix this? I looked at the previous release and it was passing.

Errors produced by Dialyzer often don't relate do the exact place, that causes the problem, so it still might be caused by your changes. 

Dialyzer fail here might be also caused by Elixir version bump, so I recommend you running Dialyzer on current `master` with Elixir 1.17.2, to check it

File: lib/membrane_opus/encoder.ex [hunk 1]
```diff
@@ -173,7 +171,29 @@ defmodule Membrane.Opus.Encoder do
   end
 
   defp set_current_pts(%{queue: <<>>} = state, input_pts) do
-    %{state | current_pts: input_pts}
+    if state.current_pts != nil and input_pts != nil and state.current_pts > input_pts do
+      diff = (state.current_pts - input_pts) |> Time.nanoseconds()
```
Comment (FelonEkonom) [inline lib/membrane_opus/encoder.ex:175]: if `state.current_pts` and `input_pts` are `Membrane.Time.t()`, there is no need to pass the difference between them to `Membrane.Time.nanoseconds/1`

File: lib/membrane_opus/encoder.ex [hunk 2]
```diff
@@ -173,7 +171,29 @@ defmodule Membrane.Opus.Encoder do
...
+      cond do
+        diff > Time.milliseconds(100) ->
+          Membrane.Logger.warning(
+            "Input buffer PTS is #{Time.pretty_duration(diff)} (#{ms} ms) smaller than expected value"
+          )
+
+        diff > Time.milliseconds(10) ->
+          Membrane.Logger.debug(
+            "Input buffer PTS is #{Time.pretty_duration(diff)} (#{ms} ms) smaller than expected value"
+          )
```
Comment (FelonEkonom) [inline lib/membrane_opus/encoder.ex:187]: Print diff only one (or in `ms`, or as result of `Time.pretty_duration`)
Add to the log expected value (`state.current_pts`) and value that occurs in the input buffer

**Review by FelonEkonom — CHANGES_REQUESTED** [review]

**Comment by mat-hek** [comment]
@FelonEkonom it seems it helps, as when I updated MP4 and Opus in boombox, tests passed, though I get a bunch of 
```
12:30:22.171 [warning] <0.672.0>/:webrtc_out_opus_encoder PTS values are overlapping
```

To reproduce, get boombox from `update-deps` branch and run `mix test` or `mix test --only webrtc_audio`

**Comment by FelonEkonom** [comment]
@mat-hek such a branch doesn't exist in `boombox`

**Review by FelonEkonom — APPROVED** [review]
@bartkrak according to what @mat-hek wrote, let's merge it as it is now and create an issue describing problem with warnings that occurs in boombox with opus plugin

File: lib/membrane_opus/util.ex [hunk 3]
```diff
@@ -69,17 +70,27 @@ defmodule Membrane.Opus.Util do
...
 
-      Enum.at(packets, 1).pts > input_pts ->
-        Membrane.Logger.warning("PTS values are overlapping")
-        :ok
+    if output_pts == nil or input_pts == nil do
+      :ok
+    else
+      # Diff constantly oscillates between approximately 2 ms - 22 ms.
+      diff = output_pts - input_pts
+      epsilon = 30 |> milliseconds()
```
Comment (mat-hek) [inline lib/membrane_opus/util.ex:80]: As we spoke, the actual overlap is no more than 2ms. Try refactoring it so that we operate on the actual overlap.

**Review by mat-hek — CHANGES_REQUESTED** [review]

**Review by mat-hek — APPROVED** [review]

**Pull request merged**